src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+
+- ``-n`` or ``-g``: the name of the resource group
+- ``-l``: the Azure Location
+
+Fortunately we have already configured our default Location value of ``eastus`` so all we need to supply is the resource group name. We will use a consistent naming convention to make sure none of our resources have conflicting names. The convention for resource groups we will use is "your name" + "cli-wt" (CLI walkthrough) + "rg" (resource group):
+
+    <name>-cli-wt-rg
+
+Note that there is a 23 character limit for resource names. If you have a long first name consider using a short name like Pat if your name is Patrick.
+
+Provisioning
+------------
+
+Now that we have determined the command structure and its arguments we can create our resource group:
+
+.. sourcecode:: powershell
+
+    > az group create -n <name>-cli-wt-rg
+
+You should see a JSON output like this:
+
+.. sourcecode:: bash
+
+    {
+        "id": "/subscriptions/<subscription ID>/resourceGroups/<name>-cli-wt-rg",
+        "location": "eastus",
+        "managedBy": null,
+        "name": "<name>-cli-wt-rg",
+        "properties": {
+            "provisioningState": "Succeeded"
+        },
+        "tags": null,
+        "type": "Microsoft.Resources/resourceGroups"
+    }
+
+Notice how the subscription and location are set automatically. The former by logging in and the latter by configuring its default value.
+
+Configuring
+-----------
+
+Just as we set a default location we will assign this resource group as a default as well. Be sure to enter your new resource group name as the value:
+
+.. sourcecode:: powershell
+
+    > az configure --default group=<name>-cli-wt-rg
+
+You can confirm the default has been set by checking the CLI configuration with the ``-l`` (list) argument and seeing that the "group" has value has been set correctly:
+
+.. sourcecode:: powershell
+
+    > az configure -l
+
+Virtual Machines
+================
+
+For this walkthrough we will not be using our VM to deploy an application but simply to get comfortable using the CLI. 
+
+Exploration
+-----------
+
+Virtual Machines are naturally more complex to interact with than a simple resource group. However, now that we understand the pattern of the ``az CLI`` that complexity can be managed using the help command to methodically work our way through its Sub-Groups and Commands.
+
+Once again let's begin by assessing what is available to us:
+
+.. sourcecode:: powershell
+
+    > az vm -h
+
+Planning
+--------
+
+Creating a VM will naturally require many Arguments to customize it. Recall in the web portal how there were several menus we had to work through to provision it. In addition to all of those options the ``az CLI`` exposes additional configuration Arguments for more granular control. 
+
+Let's see what Arguments are associated with creating a VM:
+
+.. sourcecode:: powershell
+
+    > az vm create -h
+
+From this long list of arguments we will need to provide values for the following:
```

Comment:
should arguments be capitalized here? It should at least be consistent throughout the article.